File: 103-2_iOS程式設計/SliderDemo/SliderDemo/ViewController.m
//
//  ViewController.m
//  SliderDemo
//
//  Created by CSIEVNUT403 on 2015/3/23.
//  Copyright (c) 2015年 CSIEVNUT403. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@property (strong, nonatomic) IBOutlet UILabel *label;

@property (strong, nonatomic) IBOutlet UISlider *slider;

- (IBAction)onSliderValueChanged:(id)sender;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    [self.slider setValue:10];
    [self.label setText:@"10"];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (IBAction)onSliderValueChanged:(id)sender {
    UISlider *slider = (UISlider *) sender;
    NSString *slderValue = [[NSString alloc] initWithFormat:@"%d", (int)[slider value]];
    
    [self.label setText:slderValue];
    NSLog(@"slider value = %@", slderValue);
}

@end

File: 106-2_iOS程式設計/MultiButton/MultiButton/ViewController.swift
//
//  ViewController.swift
//  MultiButton
//
//  Created by Brian Wang on 2018/5/3.
//  Copyright © 2018年 Brian Wang. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    @IBOutlet weak var labelMsg: UILabel!
    
    @IBOutlet weak var labelTel: UILabel!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
        labelTel.text = ""
        labelMsg.text = ""
    }
    
    @IBAction func numberClick(_ sender: UIButton) {
        labelTel.text = labelTel.text! + sender.currentTitle!
//        labelTel.text! += sender.currentTitle!
        
        labelMsg.text = ""
    }
    
    @IBAction func clearClick(_ sender: UIButton) {
        labelTel.text = ""
        labelMsg.text = ""
    }
    
    @IBAction func sureClick(_ sender: UIButton) {
        if labelTel.text?.lengthOfBytes(using: String.Encoding.utf8) == 10 {
            labelMsg.text = "撥打電話：" + labelTel.text!
        } else {
            labelMsg.text = "必須輸入10碼電話號碼！"
        }
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

